### PR TITLE
Update YCM and always use Eigen3 targets

### PR DESCRIPTION
**Pull request by drdanz**
Latest FindEigen3 supports the Eigen3::eigen target even for eigen < 3.3.1
Also it supports looking for the `Eigen3Config.cmake` file, therefore the switch in the YARPFindDependencies should no longer be necessary.

@traversaro, @PeterBowman please test YARP and dependencies, because I don't have a system with eigen < 3.3.1 (even though travis should be using this)

### CODE REVIEW

File: extern/ycm/CMakeLists.txt [hunk 1]
```diff
@@ -86,7 +86,8 @@ set(YCM_FOUND ${YCM_FOUND} PARENT_SCOPE)
 foreach(_d ycm-0.6.0   # since YARP 3.0.0
            ycm-0.7.0
            ycm-0.8.0
-           ycm-0.8.1)  # since YARP 3.0.1
+           ycm-0.8.1   # since YARP 3.0.1
+           ycm-0.8.20180628.11
```
Comment (PeterBowman) [inline extern/ycm/CMakeLists.txt]: Missing closing `)`.

**Comment by traversaro** [comment]
I am confused. The new  FindEigen3.cmake is used only if YCM is compiled using `YCM_MANTAINER_MODE` (see https://github.com/robotology/ycm/blob/61853801c057625261324c6efd471c6bdb57affb/3rdparty/CMakeLists.txt#L23) so by default on Ubuntu Xenial it will not be used, right?

**Comment by PeterBowman** [comment]
`if(NOT EXISTS "${CMAKE_ROOT}/FindEigen3.cmake" OR YCM_MAINTAINER_MODE)`

The first condition resolves to `/usr/share/cmake-x.x/FindEigen3.cmake` (`x.x`: current CMake version):

* There is no such file on my system (16.04, CMake 3.5), which means that the YCM file is used instead.
* This path is probably wrong - it should be `${CMAKE_ROOT}/Modules` - and it wouldn't make any difference on Xenial as libeigen3-dev already used config modules (i.e. Eigen3Config.cmake, [ref](https://packages.ubuntu.com/xenial/all/libeigen3-dev/filelist)). Same for Trusty, altough apt will install a `/usr/share/cmake-2.8/Modules/FindEigen3.cmake` file ([ref](https://packages.ubuntu.com/trusty/all/libeigen3-dev/filelist)), CMake  - as required by YARP, that is, CMake 3.5 (from the `cmake3` apt package) - should ignore it.

File: extern/ycm/CMakeLists.txt (hunk 1 above)
Comment (drdanz) [reply extern/ycm/CMakeLists.txt]: Sorry, I forgot to fix the commit :sweat_smile:
It should compile now...

**Comment by drdanz** [comment]
> The new FindEigen3.cmake is used only if YCM is compiled using YCM_MANTAINER_MODE [...] so by default on Ubuntu Xenial it will not be used, right?

If `YCM_MANTAINER_MODE` all the files are always downloaded. In this case, as @PeterBowman wrote, there used to be a `FindElgen3.cmake` file installed by the `libeigen3-dev` package, in `/usr/share/cmake-2.8/Modules` that at the time was equivalent to `${CMAKE_ROOT}/Modules`, At the time I believed that, if that file was available, it was better to use that instead of our. With the latest updates, even if CMake was updated, that file stayed in the `cmake-2.8` directory, and now it is no longer installed, therefore I'm removing that check in YCM

**Comment by drdanz** [comment]
Appveyor failure is unrelated. I'm merging this, please open an issue if this is causing failures somewhere

**Pull request merged**